src/main/scala/org/ergoplatform/nodeView/history/storage/modifierprocessors/HeadersProcessor.scala

```diff
@@ -66,6 +66,18 @@ trait HeadersProcessor extends ToDownloadProcessor with ScorexLogging with Score
     */
   def headersHeight: Int = bestHeaderIdOpt.flatMap(id => heightOf(id)).getOrElse(-1)
 
+  /**
+    * Common block between bestHeader and bestFullBlock
+    */
+  def commonBlockIdOpt: Option[ModifierId] = {
+    bestFullBlockOpt.flatMap { fb =>
+        headerChainBack(config.blocksToKeep,
```

Comment:
Why this limit? A common header could be deeper than that.
An independent question is what could be lowest limit on config.blocksToKeep (I propose to set it to be config.keepVersions at least).

Reply:
commonBlockIdOpt implementation is utterly inefficient. For every call a lot of disk reads will be done potentially. Isn't it better to store and update a variable storing the reference to the common header?